neuralpp/symbolic/eliminator.py

```diff
@@ -0,0 +1,85 @@
+from .expression import QuantifierExpression, Context, Expression, Constant, AbelianOperation, Variable, \
+    FunctionApplication
+from .basic_expression import BasicQuantifierExpression
+from .sympy_expression import SymPyExpression
+from .interval import ClosedInterval, DottedIntervals, from_constraint
+from .constants import if_then_else
+import neuralpp.symbolic.functions as functions
+from functools import reduce
+from typing import Optional, Callable
+import operator
+
+
+def _eliminate_interval(operation: AbelianOperation, index: Variable, interval: ClosedInterval, body: Expression) \
+        -> Expression:
+    # try to solve symbolically
+    result = _symbolically_eliminate(operation, index, interval, body)
+    if result is not None:
+        return result
+
+    # TODO: enable this
+    # if isinstance(interval.lower_bound, Constant) and isinstance(interval.upper_bound, Constant):
+    #     # iterate through the interval if we can
+    #     return reduce(operation,
+    #                   map(lambda num: body.replace(interval.index, Constant(num)), iter(interval)),
+    #                   operation.identity)
+
+    return BasicQuantifierExpression(operation, index, interval.to_context(index), body)
```

Comment:
Kind of. Though that's not exactly the same, since `from_constraint` (supposedly if not currently) returns several intervals in an if-then-else tree, and here we only convert to context one interval (or `DottedInterval`).